### PR TITLE
Support IPv6 in coverage collection

### LINKED ISSUE #7366: Code coverage compatibility with Dart SDK 1.22
With Dart SDK 1.22, the VM service protocol has added a security token in its URIs, so that observatory URIs now look like `http://127.0.0.1:54804/PTwjm8Ii8qg=/` rather than just a host port pair.

We're updating `package:coverage` APIs with a breaking change to take a base URI rather than host and port parameters which will be released in version 0.9.0.

On the Flutter side, we'll need a small patch to the platform plugin code coverage collection logic once the new version of `package:coverage` is released.

[Related announcement](https://groups.google.com/a/dartlang.org/forum/#!msg/announce/VxSw-V5tx8k/wPV0GfX7BwAJ)

### PR DESCRIPTION
**Pull request by tvolkert**
Builds on engine fixes to #9813

Also fixes #7366

### CODE REVIEW

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Comment by Hixie** [comment]
Test?

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -488,16 +490,17 @@ void main() {
       'FLUTTER_TEST': 'true',
       'FONTCONFIG_FILE': _fontConfigFile.path,
     };
+    print('$command');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: this is what is causing travis to fail

Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Oops - an errant debugging stmt.  Removed.

**Pull request merged**